tasks/libs/ciproviders/gitlab_api.py

```diff
@@ -610,6 +613,14 @@ def flatten(yml):
     return flatten(yml)
 
 
+def is_leaf_job(job_name, job_contents):
+    """
+    A 'leaf' job is a job that will be executed by gitlab-ci, that is a job that is not meant to be only extended (usually jobs starting with '.') or special gitlab objects (variables, stages...)
+    """
+
+    return not job_name.startswith('.') and ('script' in job_contents or 'trigger' in job_contents)
```

Comment:
Do you really need the second part of the condition? I think that any job defined in the gitlab-ci.yml that does not start by `.` will be executed by Gitlab

Reply:
Ah yeah forgot about gitlab special parameters.